spring-integration-xmpp/src/main/java/org/springframework/integration/xmpp/config/XmppConnectionFactoryBean.java

```diff
@@ -67,7 +66,7 @@
 
 	private volatile boolean running;
 
-	private volatile XMPPTCPConnection connection;
+	protected volatile XMPPTCPConnection connection;
```

Comment:
That isn't good to allow to mutate the property from the iheritors. Not safe.
If you still think we need to provide access to this property, then add `protected` getter.

Reply:
I know. The issue is that the start() method of the factory bean can not be tested at all in a unit test in its current state, because there is no way to inject a mock for the connection. The only way to test it is by either injecting a mock by reflection, adding a setter, changing the accessor (all dirty hacks), or by introducing a mock XMPP server. Is there something like the MockRestServiceServer (spring-test) for XMPP? Then we could test the interaction, this would be the preferred and best option. If there is none I guess I have to revert all changes except for the changes in the start method. And we go from there. The other option would be to add a story to add a Mock XMPP server, and after that one is done start work on this ticket again.

Reply:
Well, you are almost there! 😄 
We really use a compromise in our tests via reflection.
For this purpose we provide `TestUtils.getPropertyValue()` utility:
```
TestUtils.getPropertyValue(xmppConnectionFactoryBean, "connection", XMPPTCPConnection.class);
```
So, you `createInstance()` becomes much simpler:
```
@Override
protected XMPPConnection createInstance() throws Exception {
			return mock(XMPPTCPConnection.class);
}
```